src/menu.scss

```diff
@@ -5,165 +5,303 @@
-    padding-left: 0;
+    max-width: 20rem;
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: column;
+    -webkit-box-direction: normal;
+    -webkit-box-orient: vertical;
   }
 
-  &__list {
+  &__sublist {
     @include fd-reset();
 
-    margin: 0;
-    padding: 0;
-    list-style: none;
-    // modified with separator border
-    @at-root {
-      &--separated {
-        & > *,
-        .#{$block}__list > * {
-          border-top-style: solid;
-          border-top-width: 1px;
-
-          @include fd-var-color("border-top-color", $fd-menu-item-separator-color, --fd-menu-item-separator-color);
-
-          &:first-child {
-            border-top-width: 0;
-          }
-        }
-      }
+    left: 100%;
+    right: initial;
+    position: absolute;
+    top: 0;
+    z-index: 2;
+    margin: 0.25rem 0 0 -0.25rem; // offset of submenu
+
+    @include fd-rtl() {
+      right: 100%;
+      left: initial;
+      margin: 0.25rem -0.25rem 0 0;
     }
   }
 
-  &__title {
-    display: block;
-    margin-bottom: 0;
-    margin-top: 0;
-    padding: $fd-menu-title-padding $fd-menu-item-padding-x;
+  &__list > li {
+    position: relative; // submenu's positioning relative to parent item
+  }
 
-    @include fd-type("-1", "normal");
+  // adjust padding of item for varying screens
+  @mixin fd-item-responsive-padding {
+    @include fd-screen(s) {
+      padding: 0.75rem 1rem 0.75rem 1rem; // line-height=20px so 2.75rem or 44px-20px=24px/2= 12px or 0.75rem each
+      height: 2.75rem;
+    }
 
-    text-transform: uppercase;
+    @include fd-screen(m) {
+      padding: 0.75rem 1rem 0.75rem 0.75rem; // line-height=20px so 2.75rem or 44px-20px=24px/2= 12px or 0.75rem each
+      height: 2.75rem;
+    }
+
+    @include fd-screen(l) {
+      padding: 0.375rem 0.75rem 0.375rem 0.5rem; // line-height=20px so 2rem or 32px-20px=12px/2= 6px or 0.375rem each
+      height: 2rem;
+    }
 
-    @include fd-var-color("color", $fd-menu-title-color, --fd-menu-title-color);
-    @include fd-var-color("background-color", $fd-menu-title-background-color, --fd-menu-title-background-color);
+    @content;
   }
 
   &__item {
     @include fd-reset();
-    @include fd-ellipsis();
 
-    display: block;
+    display: flex;
+    align-items: center;
     transition: all $fd-animation--speed ease-in;
 
     @include action-cursor();
 
-    padding: $fd-menu-item-padding-y $fd-menu-item-padding-x;
+    max-width: 20rem;
+    background-color: $fd-menu-item-background-color;
 
-    @include fd-var-color("background-color", $fd-menu-item-background-color, --fd-menu-item-background-color);
-    @include fd-var-color("color", $fd-menu-item-color, --fd-menu-item-color);
-    @include fd-weight("normal");
+    &--separated {
+      border-bottom-style: solid;
+      border-bottom-width: 0.0625rem;
+      border-bottom-color: $fd-menu-item-separator-color;
+    }
 
-    text-decoration: none;
+    @include fd-hover() {
+      .#{$block}__option-name {
```

Comment:
Could you change this element name to `__title`?

Reply:
done